packages/@aws-cdk/aws-ecs/lib/fargate/fargate-service.ts

```diff
@@ -171,10 +171,15 @@ export class FargateService extends BaseService implements IFargateService {
     }
 
     this.configureAwsVpcNetworkingWithSecurityGroups(props.cluster.vpc, props.assignPublicIp, props.vpcSubnets, securityGroups);
+  }
 
-    if (!props.taskDefinition.defaultContainer) {
-      throw new Error('A TaskDefinition must have at least one essential container');
+  protected onValidate() {
+    const validationErrors = super.onValidate();
```

Comment:
I like the change, but we're changing the way we do validations in v2, and we need new code to be compatible with the new way of doing things.

New way looks like:

```
this.node.addValidation({
   validate: () => !this.taskDefinition.defaultContainer ? ['A taskdefinition...'] : [],
});
```